pulp_container/tests/functional/conftest.py

```diff
@@ -145,6 +154,118 @@ def inspect(local_url):
     return _LocalRegistry()
 
 
+@pytest.fixture(scope="session")
+def tempdir():
+    """A temporary directory."""
+    with tempfile.TemporaryDirectory() as directory:
+        yield directory
+
+
+@pytest.fixture(scope="session")
+def gpg_metadata(tempdir):
```

Comment:
There is a pulpcore fixture now (originally partly based on this design) can you look into using it?

https://github.com/pulp/pulpcore/blob/main/pulpcore/tests/functional/gpg_ascii_armor_signing_service.py

You'd probably need to wait for https://github.com/pulp/pulpcore/pull/2913 though.